src/cpp/rtps/RTPSDomainImpl.hpp

```diff
@@ -213,11 +213,13 @@ class RTPSDomainImpl
     /**
      * Find a local-process reader.
      *
-     * @param reader_guid GUID of the local reader to search.
+     * @param [in, out] local_reader Reference to the shared pointer to be set.
+     * @param           reader_guid GUID of the local reader to search.
      *
-     * @returns A pointer to a local reader given its endpoint guid, or nullptr if not found.
+     * @returns A shared pointer to a local reader given its endpoint guid, or nullptr if not found.
```

Comment:
The method has no return value, so I suggest changing this:

```suggestion
     * @post If @c local_reader had a non-null value upon entry, it will not be modified.
     *       Otherwise, it will be set to point to a local reader whose GUID is the one given in @c reader_guid, or nullptr if not found.
```